document release process (#652)
Khan/perseus: README.md

### Working

1. `git checkout main; git pull`
2. `git checkout -b [FEATURE_BRANCH_NAME]`
3. ☢️ We don’t use deploy branches in Perseus
4. `yarn start` will start [Storybook](https://storybook.js.org/) on localhost:6006
5. Do stuff
6. `yarn test` will run [Jest](https://jestjs.io/)/[RTL](https://testing-library.com/docs/react-testing-library/intro/) tests; `yarn cypress` will run [Cypress](https://www.cypress.io/) tests
7. `yarn changeset` will walk you through creating a [changeset](https://github.com/changesets/changesets) (we generally stick to [semver](https://semver.org/))
8. ☢️ Empty changesets should be considered an exception to the rule and should generally be avoided
9. `git add` and `git commit`
10. `git pull-request` will walk you through creating a pull request

### CI/CD aka Github

1. When you create/update a PR, we run a series of checks against your code
    - Gerald requests reviewers (there’s a “perseus” user group that primary maintainers are in)
    - Linting/Types/Tests; checks to make sure code is properly covered
    - Check for a changeset

> 2. 🍀 An npm snapshot is made and can be used to check changes before merging/releasing

Reviewer: ```suggestion
2. 🍀 A [snapshot release](https://github.com/changesets/changesets/blob/main/docs/snapshot-releases.md) is made and can be used to check changes before merging/releasing
```
